crates/json-rpc/src/lib.rs

```diff
@@ -135,3 +135,14 @@ impl<T> RpcReturn for T where T: DeserializeOwned + Debug + Send + Sync + Unpin
 pub trait RpcObject: RpcParam + RpcReturn {}
 
 impl<T> RpcObject for T where T: RpcParam + RpcReturn {}
+
+/// A macro to check if a field is present in a map and set it if it is.
+#[macro_export]
+macro_rules! check_and_set_field {
```

Comment:
I wouldn't export this, you can move it into a `macros` module and `#[macro_use]` on it

Reply:
just raise a question, does this means we can only use this macro inside this crate? I want to use this macro also for the filter https://github.com/alloy-rs/alloy/blob/561d392d5c52352ffb8d3632602eb3ca21de1cee/crates/rpc-types-eth/src/filter.rs#L616-L648

Reply:
Or maybe I misunderstood before and after. Is that what you meant to express?

```rs
# lib.rs
#[module_use]
mod macros;

# macros.rs
#[macro_export]
macro_rules! check_and_set_field {
    ($map:expr, $field:expr) => {{
        if $field.is_some() {
            return Err(serde::de::Error::duplicate_field(stringify!($field)));
        }
        $field = Some($map.next_value()?);
    }};
}
```

Reply:
If you want to re use it you might want to put it in alloy-serde if it doesnt add dependencies, otherwise duplicating it is also fine